### DISCUSSION TITLE
Is there a way to call the same logic for multiple hooks?

### DISCUSSION THREAD

**Discussion by mdodell** [janko/rodauth-rails#212] [Q&A] [answered]
In my `rodauth_main` file, I noticed I am repeating myself quite a bit, i.e.:

```ruby
  after_create_account do
      merge_account
  end

  after_login do
      merge_account
  end
```

Is there a way to chain multiple of those together, so I don't have to keep redefining the same callback logic?

**Comment by janko** [comment]
Not sure what you mean by chaining, but you can define custom methods on your Rodauth object that you can call from hooks to keep things DRY:

```rb
class RodauthMain < Rodauth::Rails::Auth
  configure do
    # ...
  end

  def merge_account
    # ...
  end
end
```

**Reply by mdodell** [reply]
Yup, that's what I'm already doing.

I meant being able to do something similar to a switch statement, where you can run the same logic for multiple cases, ie:

```ruby
  after_login, after_create_account do
      merge_account
  end
```

**Reply by janko** [reply, marked as answer]
I see what you mean now. No, there isn't any built-in way to chain multiple callbacks, the only option is using a loop:

```rb
%i[after_login after_create_account].each do |hook|
  public_send(hook) { merge_account }
end
```
